src/H5Tpkg.h

```diff
@@ -291,6 +297,19 @@ typedef struct H5T_array_t {
     size_t   dim[H5S_MAX_RANK]; /* size in each dimension       */
 } H5T_array_t;
 
+/* Form (Rectangular, Polar, Exponential) of complex number */
+typedef enum H5T_complex_form_t {
+    H5T_COMPLEX_RECTANGULAR,
+    H5T_COMPLEX_POLAR,
+    H5T_COMPLEX_EXPONENTIAL,
+} H5T_complex_form_t;
+
+/* A complex number datatype */
+typedef struct H5T_complex_t {
+    H5T_complex_form_t form;        /* Form (Rectangular, Polar, Exponential) of complex number */
+    bool               homogeneous; /* Whether the parts of the complex number have the same datatype */
```

Comment:
The 'homogeneous' flag is just extra overhead, since we don't support heterogeneous complex types (and probably never will, I'm guessing).   I suggest removing it, to simplify and clarify the code.

Reply:
Sure, this was mostly thinking ahead since the related forum post already mentioned some use of heterogeneous complex types, but it's not really necessary to have around right now.